Release v0.28 RC1

dr-orlovsky opened the pull request:
We still have quite a few issues and PRs pending to be addressed/merged before full 0.28 release: see https://github.com/rust-bitcoin/rust-bitcoin/milestone/10

Most important, we have to find a way to address #777; additionally it will be desirable to get #587, #786, #776.

We also have quite of work to write CHANGELOG in #785

Nevertheless I propose to start with a `beta-1` subrelease to unlock ability for `miniscript` release and downstream testing. The Taproot API looks final, and the pending PRs are addresses internal issues/bugs which is perfectly fine for beta releases.

romanz commented:
[electrs](https://github.com/romanz/electrs/) tACK :)

apoelstra commented:
concept ACK. I'm tempted to bikeshed over "beta" vs "rc".

@tcharding what do you think?

dr-orlovsky commented:
Seems like we have all our pending PRs merged so this is indeed an RC release. Updated to the tip of the master and changed to RC in version number. I propose to merge & publish this RC1.

Kixunil reviewed (approved):
ACK 9efa180529e917587a79268cb8bfef587640c777

Super happy about this one! :)

Kixunil reviewed (approved):
ACK 1b2dbd7c083e15dc349f927014a157e72f8f1bdb

tcharding commented:
> concept ACK. I'm tempted to bikeshed over "beta" vs "rc".
> 
> @tcharding what do you think?

I don't really mind what the process is as long as I understand it. Currently I do not understand the difference between beta and rc? In [this](https://github.com/rust-bitcoin/rust-bitcoin/issues/773#issuecomment-1014272051) comment @dr-orlovsky you say we do a few beta releases _then_ an rc, whats the benefit of this as opposed to just rc releases please?

tcharding commented:
The diff uses `rc` now. Pull the trigger @apoelstra!

tcharding commented:
FWIW I've gotten the following downstream crates building.
- rust-miniscript
- bdk

Please see branch `rust-bitcoin-0.28.0-rc.1` in my trees if you are interested.

We will need #790 for rc 2.

apoelstra commented:
Tagged and published.
